app/src/androidTest/java/com/example/bigowlapp/activity/SignUpPageActivityTest.java

```diff
@@ -0,0 +1,140 @@
+package com.example.bigowlapp.activity;
+
+import android.os.SystemClock;
+import android.view.View;
+
+import androidx.lifecycle.Lifecycle;
+import androidx.test.espresso.Espresso;
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+
+import com.example.bigowlapp.R;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.viewModel.EditProfileViewModel;
+import com.example.bigowlapp.viewModel.SignUpViewModel;
+
+import org.hamcrest.Matcher;
+import org.junit.FixMethodOrder;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.Random;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.hasErrorText;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.isRoot;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.mockito.Mockito.atMostOnce;
+import static org.mockito.Mockito.verify;
+
+@RunWith(AndroidJUnit4.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
```

Comment:
Just curious what this does.

Reply:
This method test the test case alphabetically. But I think what you mention above, the CI instance are clear. My test case test directly the database that's why the test D did not passed. As I am required to test the database to make sure the login component works.